UnionPatcher/ElfFile.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Buffers.Binary;
+
+/*
+Linux ELF header refspec: https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html
+Wikipedia entry on ELF: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format
+*/
+
+namespace LBPUnion.UnionPatcher {
+    public class ElfFile {
+        private enum WordSize : byte {
+            ThirtyTwoBits = 0x01,
+            SixtyFourBits = 0x02
+        }
+
+        private enum Endianness : byte {
+            Little = 0x01,
+            Big = 0x02
+        }
+
+        private enum InstructionSetArchitecture : UInt16 {
+            PowerPC = 0x15, //64-bit PowerPC (PS3)
+            ARM = 0x28 //32-bit ARM (Vita)
+        }
+
+        public string Name { get; } = "Binary Blob";
+
+        public bool IsValid { get; }
+        public bool? Is64Bit { get; } = null;
+        public bool? IsBigEndian { get; } = null;
+        public string Architecture { get; } = null;
+
+        public byte[] Contents { get; } = null;
+
+        public ElfFile(byte[] fileContents) {
+            IsValid = fileContents[0x00..0x04].SequenceEqual(new byte[] {0x7F, (byte)'E', (byte)'L', (byte)'F'});
+            if(IsValid == false) return;
+
+            byte identClassValue = fileContents[0x04];
+            byte identDataValue = fileContents[0x05];
```

Comment:
I'd check the length too, in case of a file smaller than the header of an ELF, so something like:
```csharp
if(fileContents.Length < 5) {
    Console.WriteLine("ELF too small");
    return;
}
```

Reply:
Added a basic check for minimum filesize